src/ui/ChatWindow.ts

```diff
@@ -474,7 +472,9 @@ export default class ChatWindow {
       }).catch(err => {
          Log.warn('Error during preSendMessage pipe', err);
       });
-
+      
+      this.getTranscript().pushMessage(message);
```

Comment:
At this point it is not guaranteed that the pipe has finished (e.g. a plugin could wait for a http request to finish).